client/cleanup.cc

```diff
@@ -223,41 +236,80 @@ Cleaner::is_quota_aware() const
     {
 	snapper->prepareQuota();
     }
-    catch (const DBus::ErrorException& e)
+    catch (const QuotaException& e)
     {
 	SN_CAUGHT(e);
 
-	if (strcmp(e.name(), "error.quota") == 0)
-	{
-	    cerr << "quota not working (" << e.message() << ")" << endl;
-	    return false;
-	}
+	cerr << "quota not working (" << e.what() << ")" << endl;
+	return false;
+    }
+
+    return parameters.space_limit < 1.0;
+}
+
+
+bool
+Cleaner::is_quota_satisfied() const
+{
+    QuotaData quota_data = snapper->queryQuotaData();
+
+    double fraction = (double)(quota_data.used) / (double)(quota_data.size);
```

Comment:
Check for `quota_data.size != 0` to prevent SIGDIV?

Reply:
Yes, in theory possible. BTW fraction would likely be ```inf``` in that case. I will add a check.